src/components/Layout/Layout.component.jsx

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
-import styled from 'styled-components';
-
-const Container = styled.main`
-  width: 100%;
-  display: flex;
-  flex-direction: column;
-  justify-content: center;
-  align-items: center;
-`;
+import React, { useContext, useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import Header from '../Header';
+import { AppContext } from '../../providers/App/App.provider';
+import { Container, InputSearch } from './Layout.styled';
 
 function Layout({ children }) {
-  return <Container>{children}</Container>;
+  const { state, dispatch, fetchVideos } = useContext(AppContext);
+  const [value, setValue] = useState('wizeline');
+  const history = useHistory();
+
+  useEffect(() => {
+    if (state.videos.length === 0) {
+      fetchVideos(state.query);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Most of the times that we need to use `eslint-disable` rule it actually correct to follow that the linter recommends.  Even if we only `need` this fetch once, it is a good idea to refetch when any of the items in the suggested dependency array change.  If this is causing more fetches, then the fix would be refactoring your code in a different way.